src/analysis/seminorm.lean

```diff
@@ -153,10 +307,18 @@ instance [semiring R] [module R ℝ] [has_scalar R ℝ≥0] [is_scalar_tower R 
 -- https://github.com/leanprover-community/mathlib/pull/11329#issuecomment-1008915345
 noncomputable instance : has_sup (seminorm 𝕜 E) :=
 { sup := λ p q,
-  { to_fun := p ⊔ q,
-    triangle' := λ x y, sup_le
-      ((p.triangle x y).trans $ add_le_add le_sup_left le_sup_left)
-      ((q.triangle x y).trans $ add_le_add le_sup_right le_sup_right),
+  { to_fun  := p ⊔ q,
+    nonneg' := λ x, begin
+      simp only [pi.sup_apply, le_sup_iff],
+      exact or.intro_left _ (p.nonneg _),
+    end,
+    map_zero' := begin
+      simp only [pi.sup_apply],
+      rw [← p.map_zero, sup_eq_left, p.map_zero, q.map_zero],
+    end,
+    add_le' := λ x y, sup_le
+      ((p.add_le x y).trans $ add_le_add le_sup_left le_sup_left)
+      ((q.add_le x y).trans $ add_le_add le_sup_right le_sup_right),
```

Comment:
It looks like you lost my suggestion to use `.. (p ⊔ q).to_add_monoid_seminorm` here to discharge these three goals. It's in the comment somewhere.